test/api-test.js

```diff
@@ -31,11 +33,247 @@ function checkResult(expected, schemaName, response) {
   return response;
 }
 
+function createServer() {
+  return new Promise((resolve, reject) => {
+    const server = net.createServer();
+    let port;
+    server.on('listening', function() {
+      resolve(server);
+    });
+    server.on('error', function(error) {
+      reject(error);
+    });
+    server.listen(0, '0.0.0.0');
+  });
+}
+
+
 describe('RippleAPI', function() {
   const instructions = {maxLedgerVersionOffset: 100};
   beforeEach(setupAPI.setup);
   afterEach(setupAPI.teardown);
 
+  describe('Connection', function() {
+
+    it('connection default options', function() {
+      const connection = new utils.common.Connection('url');
+      assert.strictEqual(connection._url, 'url');
+      assert(_.isUndefined(connection._proxyURL));
+      assert(_.isUndefined(connection._authorization));
+    });
+
+    it('with proxy', function(done) {
+      createServer().then((server) => {
+        const port = server.address().port;
+        const expect = 'CONNECT localhost';
+        server.on('connection', (socket) => {
+          socket.on('data', (data) => {
+            const got = data.toString('ascii', 0, expect.length);
+            assert.strictEqual(got, expect);
+            server.close();
+            done();
+          });
+        });
+
+        const options = {
+          proxy: 'ws://localhost:' + port,
+          authorization: 'authorization',
+          trustedCertificates: 'something'
```

Comment:
It really checks nothing, just adds to coverage. And I'm not sure we need to test something here.